Bottom: 7aa2c658e8fa8152dc761e7b585775befdc71b1f
Top:    11dfc1287b91ab5d1e7a593e6dcf5a25855690a6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-06 00:01:19 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index d379a6b..910c75b 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -308,6 +308,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
